mantidproject/mantid
# SimpleAPI: an option to control storing on ADS

This PR adds one more magic keyword to function-like simple api to get a handle over storing on ADS.
By default, the outputs will be stored on ADS as before, but now this can be disabled.
Disabling this will make it possible to eliminate `self.createChildAlgorithm` multiline setups by replacing them with neater function-like calls.
As a proof-of-concept, the necessary refactor has been done for `ExtractMonitors`.
The rest of the refactoring can be done in a separate ticket.

**To test:**
Code review.
Exercise with simpleapi calls with and without the new keyword, check they behave as supposed.
Pick an instance of `self.createChildAlgorithm` in python and modify it like it is done here for `ExtractMonitors`, see if it works.



Fixes #20664 . 

**Release Notes**


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
